### ISSUE TITLE
feature request: add support for ControlSocket

### ISSUE THREAD

**Issue by nusenu** [ralphwetzel/theonionbox#13]
I'm filing this here as a github issue so I get notifications if there are any updates on this. I mentioned this previously on the tor-relays ML [1].

Rationale:
In ansible-relayor [2] I support ControlSocket only (no controlPort) for simplicity (no TCP port required, no password handling, just filesystem permissions). 

It would be great if theonionbox would support ControlSocket to connect to tor.

[1] https://lists.torproject.org/pipermail/tor-relays/2016-September/010409.html
[2] https://github.com/nusenu/ansible-relayor

**Comment by diffusae** [comment]
Yes, I also would appreciate this. ControlSockets are better to handle.
Thanks in advance

**Comment by ralphwetzel** [comment]
Hi!
Beginning with the v3.2 [releases](https://github.com/ralphwetzel/theonionbox/releases) the Box has the capability to connect to a relay via a Control Socket. You may configure this in the [config file](https://github.com/ralphwetzel/theonionbox/blob/devel/theonionbox/config/theonionbox.cfg).
Be aware that this still requires a password for login. I'll simplify this with one of the next releases.
Best regards, Ralph

**Comment by diffusae** [comment]
Hi!

Sorry, didn't see the config file or the variable is missing in the default file. I tried it first with tor_control_port, but that didn't work. Thanks a lot for your work. I looks great and the webserver also have a low CPU usage. Better than everything I've used before.

Regards,

**Comment by diffusae** [comment]
How do I configure multiple instances in this config file?
```

[TorRelays]
#Define the number of relays to monitor
#Defaults to 1 if not defined!
tor_count = 2
```

Thanks a lot!

**Comment by ralphwetzel** [comment]
This is - as stated in the config file - not yet implemented:

```
# Not yet implemented:

[TorRelays]
# Define the number of relays to monitor
# Defaults to 1 if not defined!
# tor_count = 2
```
